<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Queue Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_queue_filter">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Filter on queue name:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_queue_filter">
        <property name="enabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_clear_filter">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="0">
       <widget class="QSplitter" name="splitter_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QueueTableView" name="tableView_object">
         <property name="font">
          <font>
           <family>DejaVu Sans Mono</family>
          </font>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>17</number>
         </attribute>
        </widget>
        <widget class="QSplitter" name="splitter_1">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <widget class="QTreeView" name="treeView_objects">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>2</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>DejaVu Sans Mono</family>
           </font>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="autoExpandDelay">
           <number>-1</number>
          </property>
          <property name="uniformRowHeights">
           <bool>false</bool>
          </property>
          <property name="animated">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="headerHidden">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QTextBrowser" name="textBrowser_body">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Cantarell'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="openLinks">
           <bool>false</bool>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Localhost"/>
    <addaction name="actionOpen"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>Actions</string>
    </property>
    <addaction name="actionPause"/>
    <addaction name="actionResume"/>
    <addaction name="separator"/>
    <addaction name="actionPurge"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionCopy_Messages"/>
    <addaction name="actionTrace_a_Message"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuToolbars">
     <property name="title">
      <string>Toolbars...</string>
     </property>
     <addaction name="actionConnection_toolbar"/>
     <addaction name="actionQueue_toolbar"/>
     <addaction name="actionMessage_toolbar"/>
    </widget>
    <addaction name="menuToolbars"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_QView"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuActions"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen_Localhost">
   <property name="text">
    <string>Open localhost</string>
   </property>
   <property name="statusTip">
    <string>Open localhost</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/connect-broker.png</normaloff>:/images/connect-broker.png</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="toolTip">
    <string>Open connection to a broker</string>
   </property>
   <property name="statusTip">
    <string>Open connection to a broker</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/disconnect-broker.png</normaloff>:/images/disconnect-broker.png</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="toolTip">
    <string>Close connection to broker</string>
   </property>
   <property name="statusTip">
    <string>Close connection to broker</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="statusTip">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/pause.png</normaloff>:/images/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause queue</string>
   </property>
   <property name="iconText">
    <string>Pause</string>
   </property>
   <property name="statusTip">
    <string>Pause queue</string>
   </property>
  </action>
  <action name="actionResume">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/play.png</normaloff>:/images/play.png</iconset>
   </property>
   <property name="text">
    <string>Resume queue</string>
   </property>
   <property name="iconText">
    <string>Resume</string>
   </property>
   <property name="statusTip">
    <string>Resume queue</string>
   </property>
  </action>
  <action name="actionPurge">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/purge.png</normaloff>:/images/purge.png</iconset>
   </property>
   <property name="text">
    <string>Purge messages</string>
   </property>
   <property name="iconText">
    <string>Purge</string>
   </property>
   <property name="toolTip">
    <string>Purge top messages</string>
   </property>
   <property name="statusTip">
    <string>Purge top messages</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/delete.png</normaloff>:/images/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete message</string>
   </property>
   <property name="iconText">
    <string>Delete</string>
   </property>
   <property name="toolTip">
    <string>Delete message</string>
   </property>
   <property name="statusTip">
    <string>Delete message</string>
   </property>
  </action>
  <action name="actionCopy_Messages">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/copy.png</normaloff>:/images/copy.png</iconset>
   </property>
   <property name="text">
    <string>Copy message</string>
   </property>
   <property name="iconText">
    <string>Copy</string>
   </property>
   <property name="statusTip">
    <string>Copy message</string>
   </property>
  </action>
  <action name="actionTrace_a_Message">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="toolbar_icons.qrc">
     <normaloff>:/images/trace.png</normaloff>:/images/trace.png</iconset>
   </property>
   <property name="text">
    <string>Trace a message</string>
   </property>
   <property name="iconText">
    <string>Trace</string>
   </property>
   <property name="statusTip">
    <string>Trace a message</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionConnection_toolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Connection actions</string>
   </property>
  </action>
  <action name="actionQueue_toolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Queue actions</string>
   </property>
  </action>
  <action name="actionMessage_toolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Message actions</string>
   </property>
  </action>
  <action name="actionAbout_QView">
   <property name="text">
    <string>About QView...</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Refreshing</string>
   </property>
   <property name="toolTip">
    <string>Click to pause page updates</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QueueTableView</class>
   <extends>QTableView</extends>
   <header location="global">queuetableview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="toolbar_icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_clear_filter</sender>
   <signal>clicked()</signal>
   <receiver>lineEdit_queue_filter</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>747</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>427</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
